connector/apmconnector/transaction.go

```diff
@@ -264,12 +264,14 @@ func (transaction *Transaction) ProcessRootSpan() bool {
 		return true
 	}
 
-	err := span.Status().Code() == ptrace.StatusCodeError
-	if err {
-		transaction.IncrementErrorCount(transactionName, transactionType, span.StartTimestamp(), span.EndTimestamp())
-	}
-
-	transaction.GenerateApdexMetrics(span, err, transactionName, transactionType)
+	// TODO: Error count and Apdex are calculated from metrics now. Though, the plan is to bring the following code
+	// back for languages like Ruby that do not yet generate metric data.
```

Comment:
is there a reason we can't do this today?

Reply:
No reason. Was just going to handle it in a follow up. I plan to introduce a config option that allows users to explicitly toggle generation of these metrics from spans. In the default case we will generate the metrics from metrics except for languages like Ruby.